### ISSUE TITLE
unreleased forum categories shown in filter

### ISSUE THREAD

**Issue by bmcutler** [Submitty/Submitty#10903]
**Describe the bug**

I  have set my future homework categories to release on future dates:
<img width="568" alt="Screenshot 2024-08-31 at 1 32 42 PM" src="https://github.com/user-attachments/assets/4a9b8cc7-4efb-45f3-9e16-c950d738788b">

The unreleased categories are removed fromt the create thread list:
<img width="215" alt="Screenshot 2024-08-31 at 1 33 48 PM" src="https://github.com/user-attachments/assets/25624141-db81-4a66-b53b-8cdadfb575f1">

But they are still visible in the filter:
<img width="605" alt="Screenshot 2024-08-31 at 1 34 58 PM" src="https://github.com/user-attachments/assets/d6f28b48-f8f5-44bd-987a-40c30f0750fc">

**Expected behavior**
The filter should only contain released categories.

**Comment by itsashishrath** [comment]
Hello, I am new to Submitty and is very interested in contributing on this issue. Can you assign this issue to me?

**Comment by itsashishrath** [comment]
> **Expected behavior** The filter should only contain released categories.

Hello, I'm new to Submitty. Please excuse me if I say something that doesn't make sense

I tried to reproduce the bug, and category with the future date is indeed visible in the filters to both student and instructor, I just had a doubt , should it not be visible to students only or should it not be visible at all to anyone?

**Comment by bmcutler** [comment]
The filter should look the same to student and instructor and not contain future-release categories.

**Comment by itsashishrath** [comment]
I checked `threadpostform.twig` file, where the behavior was right. It was simply checking if the visible date is null or less than current time.
Therefore I added the same logic in `ForumBar.twig` file, I checked, it worked fine,

Here are the screenshots
Unreleased Future category
![Screenshot 2024-09-05 121233](https://github.com/user-attachments/assets/22a512fe-3ee9-4aed-88d8-82c94a4a12aa)

Now its not present in filters inside forum.
![Screenshot 2024-09-05 115243](https://github.com/user-attachments/assets/2dc03453-2d24-4012-8f88-0bf37e504436)
